Correct sv drop computation

eroussy opened the pull request:
Rewrite the functions that handles sv drop:
- Correct a bug where the sv_drop was not computed when using only 1 stream and logging only the SV 0
- Replace the homemade algorithm by a pandas function that handles alignment for us.
Also
- refactor how packets reordering is checked
- add cqfd support

eroussy committed:
sv_timestamp_analysis: reformat packets reordering check

At the beginning of the function, a check is done to ensure that both
data have the same number of streams.
It is thus possible to iterate on the stream number, which avoids a
nested for loop.

Signed-off-by: Erwann Roussy <erwann.roussy@savoirfairelinux.com>

eroussy committed:
sv_timestamp_analysis: rewrite sv drop handling function

Latency computation is straight-forward when there is no sv drop.

In case of sv drop, the two tables (publisher and subscriber) doesn't
contain the same number of data.
This commit uses the pandas dataframe merge function with the "inner"
strategy to merge the two tables. This strategy only keeps the data
presents in the two tables, thus droping the lines where the SV was
dropped.

Converting datas from numpy to pandas (or revert) takes time, so :
- pandas is only used in case of sv_drop
- in that case, pandas also compute the latency

May be pandas should be used in the entire script instead of numpy, I
don't have time to test that now.

Signed-off-by: Erwann Roussy <erwann.roussy@savoirfairelinux.com>

eroussy committed:
.cqfd: add basic cqfd support.

Signed-off-by: Erwann Roussy <erwann.roussy@savoirfairelinux.com>

Paullgk commented on .cqfd/docker/Dockerfile:
```diff
@@ -0,0 +1,17 @@
+RUN apt-get update && apt-get install -y --no-install-recommends \
+	locales \
+	python3-pandas \
+  python3-numpy \
```
indent

Paullgk commented on .cqfdrc:
```diff
@@ -0,0 +1,6 @@
+name='sv-timestamp-analysis'
+
+[build]
+command='python3 sv_timestamp_analysis.py --help'
```
What is the use of cqfd if it is only for printing help?

eroussy replied:
cqfd is used to launch the tool. The command needs to be provided by the user, I can't write a flavor for that, because I don't know the name of the sv files that will be used.
However, I cannot provide a blank command to cqfd build part, so I propose to launch the help when nothing is provided.

eroussy committed:
sv_timestamp_analysis.py: correct steam count

- The value return by the get_stream_count function was not correct.
- The text of the asciidoc block was not correct

SFL: #15931
Signed-off-by: Erwann Roussy <erwann.roussy@savoirfairelinux.com>
